community-operators/descheduler/descheduler.v0.0.1.clusterserviceversion.yaml

```diff
@@ -127,7 +159,7 @@ spec:
                 - name: WATCH_NAMESPACE
                   valueFrom:
                     fieldRef:
-                      fieldPath: metadata.namespace
+                      fieldPath: metadata.annotations['olm.targetNamespaces']
```

Comment:
I am sorry but I lack context here, where is this `olm.targetNamespaces` set?

Reply:
It is set by OLM during installation. It's how OLM sets which namespaces an operator should watch.

Reply:
@ravisantoshgudimetla As long as the operator is installed via OLM, this annotation will be set appropriately. `metadata.namespace` won't work in the AllNamespaces installMode since it will be a static single namespace regardless of the installMode selected.
`metadata.annotations['olm.targetNamespaces']` is set to the namespaces watched by the operatorGroup the operator is installed into, which will be the same as `metadata.namespace` in a SingleNamespace OperatorGroup watching its own namespace. In an AllNamespaces operatorGroup, the annotation is set to an empty string, signifying that the operator should watch all namespaces.
Is this functionality supported by the descheduler operator?

Reply:
No, unfortunately descheduler operator cannot watch all the namespaces.

Reply:
Regardless, CSVs that are used with OLM should be using this fieldRef as per convention. I can update the installModes accordingly

Reply:
@ravisantoshgudimetla Tested this change on OKD and can confirm that the descheduler operator does support the AllNamespaces installMode, and successfully watches CRs in other namespaces and creates relevant resources.

Reply:
This operator works in AllNamespaces despite an OLM issue where permissions aren't upgrades to clusterPermissions properly. This is because it already has clusterPermissions to begin with. This should be changed so that the operator has only the permissions needed for it's installation OperatorGroup. But changing this will break support for AllNamespaces at this point.
@alecmerdler @ecordell thoughts?

Reply:
[Known issue](https://jira.coreos.com/browse/OLM-939), working on a fix!

Reply:
Thanks @alecmerdler ! I will merge this PR as a fix to descheduler for now, and once the fix gets in will reduce the permissions to properly match the installMode.